core/src/main/scala/org/apache/spark/util/RpcUtils.scala

```diff
@@ -32,4 +35,24 @@ object RpcUtils {
     Utils.checkHost(driverHost, "Expected hostname")
     rpcEnv.setupEndpointRef(driverActorSystemName, RpcAddress(driverHost, driverPort), name)
   }
+
+  /** Returns the configured number of times to retry connecting */
+  def numRetries(conf: SparkConf): Int = {
+    conf.getInt("spark.rpc.num.retries", 3)
+  }
+
+  /** Returns the configured number of milliseconds to wait on each retry */
+  def retryWaitMs(conf: SparkConf): Long = {
+    conf.getLong("spark.rpc.retry.wait", 3000)
+  }
+
+  /** Returns the default Spark timeout to use for Rpc ask operations. */
+  def askTimeout(conf: SparkConf): FiniteDuration = {
+    conf.getLong("spark.rpc.askTimeout", 30) seconds
```

Comment:
So use spark.network.timeout if it presents, but still use the default value "30s", such as
```Scala
conf.getTimeAsSeconds("spark.rpc.askTimeout",  conf.get("spark.network.timeout", "30s")) seconds
```